astroid/objects.py

```diff
@@ -260,7 +260,10 @@ class PartialFunction(scoped_nodes.FunctionDef):
     def __init__(
         self, call, name=None, doc=None, lineno=None, col_offset=None, parent=None
     ):
-        super().__init__(name, doc, lineno, col_offset, parent)
+        super().__init__(name, doc, lineno, col_offset, parent=None)
```

Comment:
A typical FunctionDef will define its name in the parent scope, so in the example below, inner_func should get added to outer_func.locals, but `partial(inner_func, ...)` should not be added until an Assign node adds it.
```
def outer_func():
    def inner_func(a, b):
        return a + b
   inner_partial = partial(inner_func, 1, 2)
```

I would probably prefer if there was an interface to add self to the parent frame for LocalsDictNodeNG for ClassDef and FunctionDef that could be disabled for Partial and Lambda.

Reply:
Do you think it's worth a refactor ? :)

Reply:
Maybe; I don't quite think I'm qualified here yet to say. LocalsDictNodeNG <- Lambda <- FunctionDef <- PartialFunction is a pretty weird hierarchy.